pkg/scheduler/backend/queue/backoff_queue.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queue
+
+import (
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/backend/heap"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"k8s.io/kubernetes/pkg/scheduler/metrics"
+	"k8s.io/utils/clock"
+)
+
+// backoffQueuer is a wrapper for backoffQ related operations.
+type backoffQueuer interface {
+	shouldBeBackingoff(podInfo *framework.QueuedPodInfo) bool
```

Comment:
```suggestion
	isBackingoff(podInfo *framework.QueuedPodInfo) bool
```

or `isPodBackingoff`?

Reply:
See [comment](https://github.com/kubernetes/kubernetes/pull/130214#discussion_r1961595694) above. I named the method this way not to be confusing, i.e. to not imply the pod is already backing off in backoffQ.

Reply:
Okay, changed to `isPodBackingoff`